core/tee/tee_rpmb_fs.c

```diff
@@ -1650,7 +1650,8 @@ static TEE_Result __maybe_unused fat_entry_dir_update
 			     sizeof(struct rpmb_fat_entry);
 
 	/* Only need to write if index points to an entry in cache. */
-	if (fat_entry_buf_idx < max_cache_entries) {
+	if (fat_entry_buf_idx < fat_entry_dir->num_buffered &&
```

Comment:
Can remove the 2nd test, by construction `fat_entry_dir->num_buffered` does not exceed `CFG_RPMB_FS_CACHE_ENTRIES` (`max_cache_entries` here).

Reply:
I don't think that's true.  num_buffered cannot exceed RPMB_BUF_MAX_ENTRIES, but can exceed CFG_RPMB_FS_CACHE_ENTRIES.  As I read it, the cache is split into two sections, the first (CFG_RPMB_FS_CACHE_ENTRIES) constitutes the first set of entries from RPMB.  The second section is a temporary buffer used to read each page of CFG_RPMB_FS_RD_ENTRIES entries beyond the initial cached set as needed.   fat_entry_dir_get_next() will grow the buffer up to RPMB_BUF_MAX_ENTRIES which by definition is larger than CFG_RPMB_FS_CACHE_ENTRIES.  fat_entry_deinit() explicitly calls that out where it resizes the buffer to just CFG_RPMB_FS_CACHE_ENTRIES if it finds num_buffered greater than it.

Reply:
Ok, you're right. +1.